lib/samples/filter_building_scene_layer/filter_building_scene_layer.dart

```diff
@@ -0,0 +1,343 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:flutter/material.dart';
+
+class FilterBuildingSceneLayer extends StatefulWidget {
+  const FilterBuildingSceneLayer({super.key});
+
+  @override
+  State<FilterBuildingSceneLayer> createState() =>
+      _FilterBuildingSceneLayerState();
+}
+
+class _FilterBuildingSceneLayerState extends State<FilterBuildingSceneLayer>
+    with SampleStateSupport {
+  // Create a controller for the local scene view.
+  final _localSceneViewController = ArcGISLocalSceneView.createController();
+
+  // Building scene layer that will be filtered. Set after the WebScene is loaded.
+  late final BuildingSceneLayer _buildingSceneLayer;
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  // A listing of all floors in the building scene layer
+  var _floorList = <String>[];
+
+  // The currently selected floor.
+  var _selectedFloor = 'All';
+
+  // Flag to show or hide the settings pane.
+  var _settingsVisible = false;
+
+  // Building scene layer sublayer that contains the currently selected feature.
+  BuildingComponentSublayer? _selectedSublayer;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      // Add a local scene view to the widget tree and set a controller.
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  child: ArcGISLocalSceneView(
+                    controllerProvider: () => _localSceneViewController,
+                    onLocalSceneViewReady: onLocalSceneViewReady,
+                    onTap: onTap,
+                  ),
+                ),
+                Center(
+                  child: ElevatedButton(
```

Comment:
usually we'd have a comment describing "A button to show the building filter settings." or something.